views/pages/DashboardPage/DashboardPage.test.jsx

```diff
@@ -3,34 +3,66 @@ import { render, screen, within, waitFor } from '@testing-library/react'
 import { MemoryRouter } from 'react-router-dom'
 
 import { AuthContext, NotificationContext } from '../../contexts'
-import { createParticipantList, createUser } from '../../../test/fixtures'
+import {
+  createParticipantList,
+  createUser,
+  createChart,
+} from '../../../test/fixtures'
 
 import DashboardPage from '.'
 
 const mockUser = createUser()
 const mockParticipants = createParticipantList()
+const mockedUsers = [
+  mockUser,
+  createUser({ uid: 'foo' }),
+  createUser({ uid: 'bar' }),
+]
+const mockedCharts = [
+  createChart({ chartOwner: { display_name: 'Jane' } }),
+  createChart({
+    title: 'chart two',
+    owner: 'foo',
+    chartOwner: { display_name: 'Jack' },
+  }),
+  createChart({
+    title: 'bar chart',
+    owner: 'bar',
+    chartOwner: { display_name: 'Bert' },
+  }),
+]
 
-jest.mock('react-router-dom', () => {
-  return {
-    ...jest.requireActual('react-router-dom'),
-    useOutletContext: () => {
-      return {
-        user: mockUser
-      }
+jest.mock(
+  'react-router-dom',
+  () => {
+    return {
+      ...jest.requireActual('react-router-dom'),
+      useOutletContext: () => {
+        return {
+          user: mockUser,
+          setNotification: jest.fn(),
+          users: mockedUsers,
+        }
+      },
     }
-  }
-}, { })
+  },
+  {}
+)
 
-jest.mock('../../api', () =>{
+jest.mock('../../api', () => {
   return {
     participants: {
-      all: () => Promise.resolve(mockParticipants)
+      all: () => Promise.resolve(mockParticipants),
+    },
+    charts: {
+      chart: {
+        index: () => mockedCharts,
+      },
     },
   }
 })
 describe('Dashboard Page', () => {
-  
-  test('Dashboard Page renders', async () => {
+  const renderDashboardPage = () =>
```

Comment:
Maybe we want a generic method in the `test` dir next to `fixtures.js`? Something like `renderPage` maybe.

Reply:
This is curious to me, the pattern we have is where we define the render function in the test itself. I'm curious as to what this will look like, could use your help explaining it some more

Reply:
A function in `helpers.tsx` that looks like this: 
```javascript
const renderPage = (PageComponent) => {
  return render(
       <MemoryRouter>
         <NotificationContext.Provider value={[{}, jest.fn()]}>
           <AuthContext.Provider value={[{}, jest.fn()]}>
             <PageComponent />
           </AuthContext.Provider>
         </NotificationContext.Provider>
       </MemoryRouter>
     )
}
```

Reply:
So we don't have to keep importing `MemoryRouter`/`NotificationContext`/`AuthContext` everywhere.